.github/workflows/deploy.yml

```diff
@@ -1,39 +1,49 @@
-name: Build mkdocs and deploy to GitHub Pages
+name: Build and Deploy Docs
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - master
```

Comment:
Don't do this. The existing logic runs mkdocs as a build check to ensure that it passes before merging to master. There was an `if` in the deploy step that only runs the actual deploy when the commits are merged to master.

Reply:
reverted if removal and updated `on` accordingly